app-modules/interaction/src/Filament/Actions/DraftInteractionWithAiAction.php

```diff
@@ -0,0 +1,167 @@
+use AdvisingApp\Ai\Exceptions\MessageResponseException;
+use AdvisingApp\Interaction\Models\InteractionInitiative;
+use AdvisingApp\Ai\Settings\AiIntegratedAssistantSettings;
+
+class DraftInteractionWithAiAction extends Action
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this
+            ->label('Draft with AI Assistant')
+            ->link()
+            ->icon('heroicon-m-pencil')
+            ->modalContent(fn (Page $livewire) => view('interaction::filament.actions.draft-with-ai-modal-content', [
+                'recordTitle' => $livewire->record['full_name'],
+                'avatarUrl' => AiAssistant::query()->where('is_default', true)->first()
+                    ?->getFirstTemporaryUrl(now()->addHour(), 'avatar', 'avatar-height-250px') ?: Vite::asset('resources/images/canyon-ai-headshot.jpg'),
+            ]))
+            ->modalWidth(MaxWidth::ExtraLarge)
+            ->modalSubmitActionLabel('Draft')
+            ->form([
+                Textarea::make('instructions')
+                    ->hiddenLabel()
+                    ->rows(4)
+                    ->placeholder('What do you want to write about?')
+                    ->required(),
+            ])
+            ->action(function (array $data, Get $get, Set $set, Page $livewire) {
+                $service = Feature::active('ai-integrated-assistant-settings')
+                    ? app(AiIntegratedAssistantSettings::class)->default_model->getService()
+                    : app(AiSettings::class)->default_model->getService();
+
+                $userName = auth()->user()->name;
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $model = $livewire::getResource()::getModelLabel();
+
+                $context = collect();
+
+                if ($id = $get('interaction_initiative_id')) {
+                    if ($initiative = InteractionInitiative::find($id)) {
+                        $context->push("- Mention the related initiative: {$initiative->name}.");
+                    }
+                }
+
+                if ($id = $get('interaction_driver_id')) {
+                    if ($driver = InteractionDriver::find($id)) {
+                        $context->push("- Include the call driver: {$driver->name}.");
+                    }
+                }
+
+                if ($id = $get('interaction_outcome_id')) {
+                    if ($outcome = InteractionOutcome::find($id)) {
+                        $context->push("- State the interaction outcome: {$outcome->name}.");
+                    }
+                }
+
+                if ($id = $get('interaction_type_id')) {
+                    if ($type = InteractionType::find($id)) {
+                        $context->push("- Specify the type of engagement: {$type->name}.");
+                    }
+                }
```

Comment:
This looks great to me and works. However, I wonder if we want not to be perscriptive to the AI here.

For example just saying `- The related initiative is: {$initiative->name}.`, `- The call driver is: {$driver->name}.`, etc.

What do you think @payal-canyon and @danharrin?

Reply:
It should read like a conversation, where you are describing each of these properties to the other person and explaining it to them as if they have never heard of these concepts before. These are general language learning models, they don't know the specific terms in our domain such as "call driver" or "initiative" without explanation. Doing so will yield the most relevant completions.

Reply:
> It should read like a conversation, where you are describing each of these properties to the other person and explaining it to them as if they have never heard of these concepts before. These are general language learning models, they don't know the specific terms in our domain such as "call driver" or "initiative" without explanation. Doing so will yield the most relevant completions.

Hey @danharrin ,

How about the new prompt as per the new suggestion, I tried to describe the additional context as well 🚀.
Let me know your thoughts on this.

Thank you.

Reply:
@payal-canyon, what you did, I think, looks fine. I hesitate to describe them, though. Perhaps we should run each of these by Product in a comment of the ticket to ensure the explanations and what is being asked of the AI makes sense and would result in what Product is looking for.